Test failure is due to actor availability. Doesn't happen locally when I run the tests on Sonoma. I guess it happens because CI is using an ancient version of macOS?

```
Command Output (stderr):
--
/Users/ec2-user/jenkins/workspace/swift-PR-macos-smoke-test/branch-main/swift/test/decl/nested/protocol.swift:52:13: error: unexpected error produced: concurrency is only available in macOS 10.15.0 or newer
      actor Level3 {
            ^
/Users/ec2-user/jenkins/workspace/swift-PR-macos-smoke-test/branch-main/swift/test/decl/nested/protocol.swift:52:13: error: unexpected note produced: add @available attribute to enclosing actor
      actor Level3 {
            ^
```

Fixed by changing that actor be an enum. Added an additional test for nesting a protocol in an actor with an availability annotation.

--- a/test/Serialization/nested-protocols.swift
+++ b/test/Serialization/nested-protocols.swift
@@ -1,6 +1,6 @@
 // RUN: %empty-directory(%t)
 // RUN: %target-swift-frontend -emit-module -module-name nestedprotocolsource -emit-module-path %t/nestedprotocolsource.swiftmodule -primary-file %S/Inputs/nested-protocols.swift
-// RUN: %target-swift-frontend -emit-sil %s -I %t | Filecheck %s
+// RUN: %target-swift-frontend -emit-sil %s -I %t | %FileCheck %s
 import nestedprotocolsource
 
 // CHECK: usesNested<A>(_:)

